### PR TITLE
Format Haskell with ormolu and elm with elm-format

### PR DESCRIPTION
**Pull request by cgeorgii**
- Use `ormolu`
- Use `elm-format`
- Add info about formatting to `README`
- Add convenience scripts to format entire codebase

### CODE REVIEW

**Comment by aspiwack** [comment]
Use Ormolu, not Fourmolu. Rationale: this is what we use, and it's not a conversation that I want projects to spend time on.

File: README.md [hunk 1]
```diff
@@ -129,6 +129,12 @@ The executable accepts two options:
 - `--config`, which allows to customize the path of the configuration file
 - `--jwk`, which allows to customize the path of the file where the JWK is stored
 
+### Formatting
+The Haskell files are formatted using `fourmolu`. The configuration is found in `fourmolu.yaml`.
```
Comment (marcosh) [inline README.md:133]: shouldn't it be `ormolu`?

File: spec/InMemoryContentRepository.hs [hunk 2]
```diff
@@ -1,34 +1,32 @@
...
-import Tagger.Owned (Owned(..))
-import Tagger.Tag (Tag)
-import Tagger.User (User)
-
 -- base
 import Control.Monad.IO.Class (liftIO)
-import GHC.Conc (TVar, atomically, readTVar, writeTVar)
-import Prelude hiding (filter)
-
 -- containers
```
Comment (marcosh) [inline spec/InMemoryContentRepository.hs:5]: these comments make no sense now and should be removed manually

File: README.md [hunk 3]
```diff
@@ -129,6 +129,12 @@ The executable accepts two options:
...
+The Elm source code is formatted using `elm-format`. There is no configuration for it.
+
+There is a script to format all files in the codebase (Elm and Haskell) under `bin/format`.
```
Comment (marcosh) [inline README.md:136]: would be nice to have (also) two separate scripts for `Elm` and `Haskell`

**Review by marcosh — CHANGES_REQUESTED** [review]
I really value having the imports separated by package, with the package explicitly mentioned.

Is there a way, both for `elm-format` and `ormolu`, to specify that imports should not be touched?

**Comment by cgeorgii** [comment]
@marcosh No, unfortunately they are both opinionated and won't allow for configuration. I personally don't see the value of adding the package name to the import list, as it adds more noise than useful information.

**Comment by marcosh** [comment]
> @marcosh No, unfortunately they are both opinionated and won't allow for configuration. I personally don't see the value of adding the package name to the import list, as it adds more noise than useful information.

The value is that, when you have a bunch of imports, you can easily see from which package the values are coming from

**Pull request merged**